### PR TITLE
lz4 Support

### PR DESCRIPTION
**Pull request by sghaskell**
@emmett9001 Sorry this took so long, but here is the work I've done to add support for lz4. It requires [lz4 Bindings for Python](https://pypi.python.org/pypi/lz4). The methods [encode_lz4_old_kafka](https://github.com/Parsely/pykafka/compare/master...sghaskell:feature/lz4-support?expand=1#diff-be2d15aff9900ab3baca0523701a468eR213) and [decode_lz4_old_kafka](https://github.com/Parsely/pykafka/compare/master...sghaskell:feature/lz4-support?expand=1#diff-be2d15aff9900ab3baca0523701a468eR244) have been borrowed from [kafka-python codec.py](https://github.com/dpkp/kafka-python/blob/a00f9ead161e8b05ac953b460950e42fa0e0b7d6/kafka/codec.py). They exist to work around the bug I discuss [here](https://github.com/Parsely/pykafka/issues/709#issuecomment-324156080). In order to compress without error I had to set [block_mode=1](https://github.com/Parsely/pykafka/compare/master...sghaskell:feature/lz4-support?expand=1#diff-be2d15aff9900ab3baca0523701a468eR192) for [independent blockmode](http://python-lz4.readthedocs.io/en/latest/lz4.frame.html?highlight=block_mode#lz4.frame.compress). 

There is little to no error handling and I have not added any tests for the code.

I was able to successfully produce and consume lz4 v0 and v1 messages in my limited testing. Have a look and give it a try.

### CODE REVIEW

**Comment by emmett9001** [comment]
@sghaskell Thanks, I'll check this out in the coming days. Can you enable me to push commits to your branch so we can both contribute to this PR?

**Comment by sghaskell** [comment]
@emmett9001 I just added you as a collaborator.

File: pykafka/utils/compression.py [hunk 1]
```diff
@@ -29,11 +29,32 @@
 except ImportError:
     snappy = None
 
+try:
+    import lz4.frame as lz4
+except ImportError:
+    lz4 = None
+
+#try:
+#    import lz4f
```
Comment (emmett9001) [inline pykafka/utils/compression.py]: Can you explain what's going on here? Why might we need `lz4f` instead of `lz4`?

Comment (sghaskell) [reply pykafka/utils/compression.py]: Looking deeper into [kafka-python codec.py](https://github.com/dpkp/kafka-python/blob/a00f9ead161e8b05ac953b460950e42fa0e0b7d6/kafka/codec.py#L205-L216) it looks like it's there to support interoperable lz4 framing.

See [KIP-57](http://grokbase.com/t/kafka/dev/164sdz8cvb/kip-57-interoperable-lz4-framing)

Comment (emmett9001) [reply pykafka/utils/compression.py]: What's your intention with having it commented here? Is it a placeholder? I don't see it being used anywhere other than the check for the presence of an `lz4` library.

File: pykafka/utils/compression.py [hunk 2]
```diff
@@ -166,3 +187,93 @@ def _detect_xerial_stream(buff):
         header = struct.unpack('!' + _XERIAL_V1_FORMAT, bytes(buff)[:16])
         return header == _XERIAL_V1_HEADER
     return False
+
+
+if lz4:
+    encode_lz4 = lz4.compress  # pylint: disable-msg=no-member
```
Comment (emmett9001) [inline pykafka/utils/compression.py]: @sghaskell The tests I wrote are passing just fine without setting the `block_mode` on this call. Am I missing something?

**Comment by emmett9001** [comment]
@sghaskell I've gotten the code to a point where I think it's mergeable assuming the tests pass. I'm still working on getting Travis to run the tests (it appears to have broken somehow...) but is there anything you notice here that should be different? I've mirrored kafka-python's implementation pretty closely.

File: pykafka/utils/compression.py (hunk 2 above)
Comment (sghaskell) [reply pykafka/utils/compression.py]: I would imagine any simple tests you write will pass without setting block_mode. I found that I had to set it when producing messages due to a lack of support for dependent-block compression.

See [KIP-57](https://cwiki.apache.org/confluence/display/KAFKA/KIP-57+-+Interoperable+LZ4+Framing)

* The only flag left unsupported is dependent-block compression (LZ4 Stream API), which our implementation does not currently support.

Comment (emmett9001) [reply pykafka/utils/compression.py]: I wonder how kafka-python manages this https://github.com/dpkp/kafka-python/blob/a3995b38210d3700855d5b438529a5387c095569/kafka/codec.py#L198 cc @dpkp

Comment (emmett9001) [reply pykafka/utils/compression.py]: I've actually managed to get this working against both 0.8.2 and 0.10.1 without setting `block_mode`.

Comment (sghaskell) [reply pykafka/utils/compression.py]: Awesome! It must have been user error on my part then. Glad to hear it works!

**Pull request merged**